src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -129,6 +133,13 @@ public ITestRunRequest CreateTestRunRequest(TestRunCriteria testRunCriteria, Pro
             var testHostManager = this.testHostProviderManager.GetTestHostManagerByRunConfiguration(testRunCriteria.TestRunSettings);
             testHostManager.Initialize(TestSessionMessageLogger.Instance, testRunCriteria.TestRunSettings);
 
+            // Allow TestRuntimeProvider to update source map, this is required for remote scenarios.
+            // If we run for specific tests, then we expect the test case object to contain correct source path for remote scenario as well
```

Comment:
Is this a valid assumption? Is it because we know a `Discovery` must have preceded a `RunSelected`? What if user changes the Target Device between discovery and run selected?